cmake/EthCompilerSettings.cmake

```diff
@@ -48,14 +48,26 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	add_compile_options(-Wextra)
 	add_compile_options(-Werror)
 	add_compile_options(-pedantic)
+	add_compile_options(-Wmissing-declarations)
 	add_compile_options(-Wno-unknown-pragmas)
 	add_compile_options(-Wimplicit-fallthrough)
 	add_compile_options(-Wsign-conversion)
 	add_compile_options(-Wconversion)
+	add_compile_options(-Wcast-qual)
+	add_compile_options(-Wcast-align)
```

Comment:
Of course this cannot be so easy. These two options cause compiler issues on Windows, and the ossfuzz target.